sql/src/main/java/org/apache/druid/sql/calcite/external/ExternalDataSource.java

```diff
@@ -141,7 +141,7 @@ public DataSource withUpdatedDataSource(DataSource newSource)
   @Override
   public byte[] getCacheKey()
   {
-    return null;
+    return new byte[0];
```

Comment:
This should either not be cacheable or it needs to build up an identifier for the external data source...  `null` seems correct for now.

Reply:
So where the `isCacheable` for the data source is false, the `getCacheKey()` should return null for those data sources. Is my understanding correct ?

Reply:
Basically, yes.  I had forgotten about that method.  In some future, we should refactor to completely eliminate the method...  But let's not take that on yet.